addon/components/plaid-bar/component.js

```diff
@@ -0,0 +1,114 @@
+import Ember from 'ember';
+import GroupElement from '../../mixins/group-element';
+
+const {
+  assert,
+  Component,
+  getProperties,
+  run: { scheduleOnce },
+  typeOf
+} = Ember;
+
+const PlaidBarComponent = Component.extend(GroupElement, {
+  layout: null,
+
+  /**
+   * Represents the bar orientation. May be "vertical" or "horizontal"
+   *
+   * @public
+   * @type {String}
+   */
+  orientation: 'vertical',
+
+  /**
+    xScale function
+
+    @public
+    @type {D3 Scale}
+  */
+  xScale: null,
+
+  /**
+    yScale function
+
+    @public
+    @type {D3 Scale}
+  */
+  yScale: null,
+
+  /**
+    Values to render bars from. These should be the same as those used
+    for the domains of the scaling functions.
+
+    @public
+    @type {Array}
+  */
+  values: [],
+
+  /**
+    @private
+  */
+  drawnValues: [],
+
+  fill: 'black',
+
+  fillOpacity: 1.0,
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+
+    let orientation = this.get('orientation');
+
+    assert(`bar chart orientation must be in {vertical,horizontal}, was "${orientation}"`,
+      orientation === 'vertical' || orientation === 'horizontal');
+
+    let checkScale = orientation === 'vertical' ? 'xScale' : 'yScale';
+
+    assert(`${checkScale} must be a band-scale for ${orientation} bar charts`, typeOf(this.get(checkScale).bandwidth) === 'function');
+
+    scheduleOnce('afterRender', this, this.drawBars);
+  },
+
+  drawBars() {
+    let { values, drawnValues, xScale, yScale, fill, fillOpacity, orientation } =
+      getProperties(this, 'values', 'drawnValues', 'xScale', 'yScale', 'fill', 'fillOpacity', 'orientation');
+
+    let x, width, y, height;
+
+    if (orientation === 'vertical') {
+      let maxHeight = Math.max(...yScale.range());
```

Comment:
We should probably standardize on using `d3-array` min/max here, not that there's much efficiency gains on a 2 element array :D